model/annotations/task_annotations.go

```diff
@@ -204,6 +204,57 @@ func UpdateAnnotation(a *TaskAnnotation, userDisplayName string) error {
 	return errors.Wrapf(err, "problem adding task annotation for '%s'", a.TaskId)
 }
 
+func AddToAnnotation(a *TaskAnnotation, userDisplayName string) error {
+	existingAnnotation, err := FindOneByTaskIdAndExecution(a.TaskId, a.TaskExecution)
+	if err != nil {
+		return errors.Wrapf(err, "finding task annotation for task id '%s'", a.TaskId)
+	}
+	if existingAnnotation == nil {
+		return errors.Wrapf(err, "missing task annotation for task id '%s'", a.TaskId)
+	}
+	source := &Source{
+		Author:    userDisplayName,
+		Time:      time.Now(),
+		Requester: APIRequester,
+	}
+	update := bson.M{
+		TaskIdKey:        a.TaskId,
+		TaskExecutionKey: a.TaskExecution,
+	}
+
+	// Overwrite note and metadata.
+	if a.Note != nil {
```

Comment:
We can technically add on to the note. Should we consider doing that?

Reply:
hmm, I'm not sure. It would depend on how they are using it and if DAG is still sending the metadata in the note. This would probably be a question for Maria or Sanjog.

Reply:
@sanjogDB do we have a preference on what we want to do?

Reply:
let me understand how DAG uses it and I'll get back after that!